### PR TITLE
TINY-10978: Uploadcare changes

### PR DESCRIPTION
**Pull request by spocke**
Related Ticket: TINY-10978

Description of Changes:
* Includes all the individual PRs rebased against `main`

If you find anything odd then we log that as a follow up ticket.

- [x] Do-not squash merge

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/themes/silver/main/ts/ui/context/ContextFormGroup.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  dom: {
+    tag: 'div',
+    classes: [ 'tox-context-form__group' ]
+  },
+  components: [ ...pLabel.toArray(), pField ],
+  fieldBehaviours: Behaviour.derive([
+    Disabling.config({
+      disabled: () => providers.checkUiComponentContext('mode:design').shouldDisable,
+      onDisabled: (comp) => {
+        // TODO: Is this really the best way to move focus out of the input when it gets disabled
```
Comment (TheSpyder) [inline modules/tinymce/src/themes/silver/main/ts/ui/context/ContextFormGroup.ts:21]: Is there a ticket for this?

**Review by TheSpyder — APPROVED** [review]

File: .changes/unreleased/tinymce-TINY-11494-2024-11-07.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+project: tinymce
+kind: Added
+body: New `onSetup` function to the context forms.
```
[the remark this answers is not in the document]
Comment (spocke) [reply .changes/unreleased/tinymce-TINY-11494-2024-11-07.yaml:3]: Not so sure about these I think it looks a bit weird to have the `something` at the start of the line. I've seen taht we mostly use `New` or `Added` in that category as the first word. The `Added` prefix under a `Added` category is a bit strange though.

File: modules/tinymce/src/themes/silver/main/ts/ui/context/ContextFormGroup.ts (hunk 1 above)
Comment (spocke) [reply modules/tinymce/src/themes/silver/main/ts/ui/context/ContextFormGroup.ts]: Added a ticket reference to this one.

File: .changes/unreleased/tinymce-TINY-11494-2024-11-07.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (spocke) [reply .changes/unreleased/tinymce-TINY-11494-2024-11-07.yaml]: I think we can discuss this a bit how we want these to work. Maybe even have some guidelines in some document it's totally random what we do here.

[the remark this answers is not in the document]
Comment (TheSpyder) [reply .changes/unreleased/tinymce-TINY-11494-2024-11-07.yaml]: I started a thread in slack

**Pull request merged**